File: io/lucamienert/basic/token/Token.java
package io.lucamienert.basic.token;

import java.util.Objects;
import java.util.Optional;

public class Token {
    
    private final TokenType type;
    private final Optional<String> value;

    public Token(TokenType type) {
        this.type = type;
        this.value = Optional.empty();
    }

    public Token(TokenType type, String value) {
        this.type = type;
        this.value = Optional.of(value);
    }

    public TokenType getTokenType() {
        return type;
    }

    public Optional<String> getValue() {
        return value;
    }

}

File: src/BasicCompiler/Token/Token.cs
using BasicCompiler.Helper;

namespace BasicCompiler.Token;

public class Token
{
    public TokenType Type { get; }
    public Optional<string> Value { get; }

    public Token(TokenType type)
    {
        Type = type;
        Value = Optional<string>.Empty();
    }

    public Token(TokenType type, string value)
    {
        Type = type;
        Value = Optional<string>.Of(value);
    }
}
